js/freewheel.js

```diff
@@ -261,9 +261,31 @@ OO.Ads.manager(function(_, $) {
         ].join('')
       ];
 
-      // TODO: Implement a timeout here in case fw_onAdRequestComplete doesn't get called - PBI-731
+      // Set up function to log error if ad response is null after maximum duration allowed for ad request to respond
+      // has exceeded.
+      var adRequestTimeout = _.bind(function(){
+        if (!fwContext._adResponse) {
+          var error = "ad request timeout";
+          OO.log("FW: freewheel ad request timeout");
+          amc.raiseAdError("FW: An ad error has occurred. The error string reported was: " + error);
+          slotEndedCallbacks[adRequestType]();
+          delete slotEndedCallbacks[adRequestType];
+        }
+      }, this);
+
       fwContext.submitRequest();
       fwAdDataRequested = true;
+      setAdRequestTimeout(adRequestTimeout, amc.MAX_AD_REQUEST_TIMEOUT);
```

Comment:
If the adRequestTimeout already exists, then we are in a bad state.  So throw error.